transformer_engine/jax/cpp_extensions/attention.py

```diff
@@ -1411,6 +1414,529 @@ def _cross_attn_bwd(
 register_primitive(FusedAttnCPWithAllGatherBwdPrimitive)
 
 
+@dataclass(frozen=True)
+class _FusedAttnCPWithP2PHelper:
+    """Helper class to assist with running the P2P ring strategy for CP attention."""
+
+    mesh: jax.sharding.Mesh
+    config: _FusedAttnConfig
+
+    @staticmethod
+    @cache
+    def use_scanloop():
```

Comment:
This is a bit ugly but a necessity of using scan loop which is preferred as it gives XLA more control over unrolling. We won't prevent not using scan but it currently warns the user to update their XLA flags. `NVTE_FUSED_RING_ATTENTION_USE_SCAN` if undefined will default to 1.

Hopefully `--xla_experimental_ignore_channel_id` becomes default in XLA:GPU at some point.